src/test/java/ApiTests.java

```diff
@@ -0,0 +1,217 @@
+import auth.login.PostUserCredentialsMethod;
+import com.zebrunner.agent.core.annotation.TestCaseKey;
+import com.zebrunner.carina.api.APIMethodPoller;
+import com.zebrunner.carina.api.apitools.validation.JsonComparatorContext;
+import com.zebrunner.carina.api.http.HttpResponseStatusType;
+import com.zebrunner.carina.core.IAbstractTest;
+import com.zebrunner.carina.utils.R;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import model.DeletePostRs;
+import model.GetAllPostsRs;
+import model.PostRs;
+import model.SearchPostsRs;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+import posts.*;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ApiTests implements IAbstractTest {
+    private String userId;
+
+    @BeforeClass
+    public void before() {
+        PostUserCredentialsMethod postCreds = new PostUserCredentialsMethod();
+        JsonPath jsonPath = postCreds.callAPI().jsonPath();
+        userId = jsonPath.getString("id");
+    }
+
+    @DataProvider(name = "get incorrect user ids")
+    public Object[][] getIncorrectUserIds() {
```

Comment:
I suggest to make same names to avoid confusions later
```suggestion
    @DataProvider(name = "incorrectUserIds")
    public Object[][] incorrectUserIds() {
```